### PR TITLE
Get haret to be clippy-clean, and check clippy on travis builds

### PR DESCRIPTION
**Pull request by erickt**
This disables a few lints, that I felt were a little unnecessary, and
could be addressed later:

* [large_enum_variant](https://github.com/Manishearth/rust-clippy/wiki#large_enum_variant)
* [collapsible_if](https://github.com/Manishearth/rust-clippy/wiki#collapsible_if)
* [while_let_on_iterator](https://github.com/Manishearth/rust-clippy/wiki#while_let_on_iterator)

### CODE REVIEW

File: haret/src/vr/states/common/normal.rs [hunk 1]
```diff
@@ -67,12 +67,11 @@ pub fn handle_start_view<T: State>(state: T,
 }
 
 pub fn handle_get_state<T: State>(state: T,
-                                  msg: GetState,
+                                  GetState { epoch, view, op }: GetState,
```
Comment (andrewjstone) [inline haret/src/vr/states/common/normal.rs]: woah, Rust allows pattern matching in fn args now? Def going to take advantage of that.

**Comment by andrewjstone** [comment]
@erickt For some reason clippy isn't being found by travis. Is there anything I can do to help  ?

File: haret/src/vr/states/common/normal.rs (hunk 1 above)
Comment (jrgarcia) [inline haret/src/vr/states/common/normal.rs]: 😲 I didn't know this either. What is this?! Erlang?!

Comment (mrallen1) [inline haret/src/vr/states/common/normal.rs]: metajack talked to me about rust in 2012 and said, "it's the semantics of Erlang with the speed of C" and I was like "sold!" soooo, maybe pattern matching has been lurking around?

Comment (andrewjstone) [inline haret/src/vr/states/common/normal.rs]: Lol, I wouldn't go that far @mrallen1 . Pattern matching has existed via `match` statements forever, but I didn't realize it was also possible in fn clauses.

Comment (erickt) [inline haret/src/vr/states/common/normal.rs]: We've had matching in fn args for years! I just checked, and we have one test for this from 2013 :)

Comment (erickt) [inline haret/src/vr/states/common/normal.rs]: @mrallen1 we used to have Erlang-style threads, with supervisor notifications when threads died, but we unfortunately ripped that out a long time ago. It imposed a bit thicker of a runtime than what we were aiming for.

Comment (andrewjstone) [inline haret/src/vr/states/common/normal.rs]: Wow, I swear I tried this a couple years ago and failed. I must have "Done it Wrong"™

**Comment by erickt** [comment]
@andrewjstone: My fault. I pushed up what I hope is a fix for clippy.

**Comment by andrewjstone** [comment]
Great. Thanks!

**Pull request merged**